### PR TITLE
Add machine learning serving example using python

### PR DESCRIPTION
**Pull request by yubozhao**
Add the code sample of deploying ML model server built with BentoML to Knative.

This example walks through building a classifier model using the iris dataset, saving the model with BentoML and then deploy the model as API server to Knative.

### Proposed Changes

* Add tutorial on how to deploy a machine learning model to Knative Serving with BentoML

### CODE REVIEW

**Comment by yubozhao** [comment]
@googlebot I signed it!

File: community/samples/serving/helloworld-python-bentoml/README.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+## Recreating sample code
+
+1. BentoML creates a model API serving, via prediction service abstraction. This code
+  defines a prediction service that requires a scikit-learn model. It asks BentoML to
+  figure out the required pip dependencies, also defined an API, which is the entry
+  point for accessing this machine learning service. Save the
+  following code into a file named `iris_classifier.py`:
+
+    ```python
```
Comment (evankanderson) [inline community/samples/serving/helloworld-python-bentoml/README.md:44]: I'd suggest putting these files into the same directory, and then including them here with the following shortcode (will work on the website which is rendered by hugo, but not in github markdown): `{{% readfile file="iris_classifier.py" %}}`

That way, people can also check out the files directly from GitHub if they don't want to copy-paste.

File: community/samples/serving/helloworld-python-bentoml/README.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+  and a Docker Hub account configured. Docker Hub will be used for a container registry).
+- Python 3.6 or above installed and running on your local machine.
+  - `scikit-learn` and `bentoml` packages, run command:
+
+    ```shell
+    pip install scikit-learn
+    pip install bentoml
+    ```
+
+## Recreating sample code
```
Comment (evankanderson) [inline community/samples/serving/helloworld-python-bentoml/README.md:36]: Put some comment in here about running all of these steps locally, just so it's clear that these don't need a kubernetes cluster.

File: community/samples/serving/helloworld-python-bentoml/README.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+    curl -v -i \
+    --header "Content-Type: application/json" \
+    --request POST \
+    --data '[[5.1, 3.5, 1.4, 0.2]]' \
+    127.0.0.1:5000/predict
+    ```
+
+## Building and deploying the sample
+
+BentoML supports containerlization a dockerfile for API server of the saved model.
```
Comment (evankanderson) [inline community/samples/serving/helloworld-python-bentoml/README.md:156]:
```suggestion
BentoML supports containerization using a Dockerfile for API server of the saved model.
```

I don't see where the Dockerfile is created here?

File: community/samples/serving/helloworld-python-bentoml/README.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+                periodSeconds: 5
+                failureThreshold: 3
+                timeoutSeconds: 60
+    ```
+
+    ```shell
+    kubectl apply --filename service.yaml
+    ```
+
+    Now that your service is created, Knative performs the following steps:
```
Comment (evankanderson) [inline community/samples/serving/helloworld-python-bentoml/README.md:207]:
```suggestion
3. Now that your service is created, Knative performs the following steps:
```

The other tutorials have this as a separate numbered bullet point.

**Comment by evankanderson** [comment]
One other note (also left on slack):

I don’t know what the “iris” dataset is, and [the scikit website isn’t clear](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_iris.html), either. It might be worth spelling out a bit about the data set and how the predictions work.

File: community/samples/serving/machinelearning-python-bentoml/iris_classifier.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+from bentoml.handlers import DataframeHandler
+from bentoml.artifact import SklearnModelArtifact
+
+@env(auto_pip_dependencies=True)
+@artifacts([SklearnModelArtifact('model')])
+class IrisClassifier(BentoService):
+
+    @api(DataframeHandler)
+    def predict(self, df):
+        return self.artifacts.model.predict(df)
```
Comment (mattmoor) [inline community/samples/serving/machinelearning-python-bentoml/iris_classifier.py:11]: Add trailing newline:
```suggestion
        return self.artifacts.model.predict(df)

```

File: community/samples/serving/machinelearning-python-bentoml/main.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    clf.fit(X, y)
+
+    # Create a iris classifier service instance
+    iris_classifier_service = IrisClassifier()
+
+    # Pack the newly trained model artifact
+    iris_classifier_service.pack('model', clf)
+
+    # Save the prediction service to disk for model serving
+    saved_path = iris_classifier_service.save()
```
Comment (mattmoor) [inline community/samples/serving/machinelearning-python-bentoml/main.py:22]: Add trailing newline:
```suggestion
    saved_path = iris_classifier_service.save()

```

File: community/samples/serving/machinelearning-python-bentoml/service.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+              path: /healthz
+            initialDelaySeconds: 3
+            periodSeconds: 5
+          readinessProbe:
+            httpGet:
+              path: /healthz
+            initialDelaySeconds: 3
+            periodSeconds: 5
+            failureThreshold: 3
+            timeoutSeconds: 60
```
Comment (mattmoor) [inline community/samples/serving/machinelearning-python-bentoml/service.yaml:24]: Add trailing newline:
```suggestion
            timeoutSeconds: 60

```

File: community/samples/serving/helloworld-python-bentoml/README.md (hunk 3 above)
Comment (yubozhao) [reply community/samples/serving/helloworld-python-bentoml/README.md]: Dockerfile is automatically generated as part of the saving model process.  I updated the readme to make it clear.  Thank you for the suggestion

File: community/samples/serving/helloworld-python-bentoml/README.md (hunk 1 above)
Comment (yubozhao) [reply community/samples/serving/helloworld-python-bentoml/README.md]: Great suggestion, I remove the code blocks into actual files.

**Comment by evankanderson** [comment]
>      community/samples/README.md
>    	ERROR	https://github.com/knative/community/tree/master/DOCS-CONTRIBUTING.md
>     		Not Found (HTTP error 404)

I'll send a PR to clean up that link.

**Pull request merged**